From 8b8c4c2fbc2f7b12f5a05ec4cec4d3749bf061ca Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 4 Dec 2024 16:45:30 +0800
Subject: [PATCH 203/301] AOSCOS: platform: lemote3a-laptop: fix
 pci_enable_device() usage

pci_enable_device() ignores return values.

Follow other use cases in the kernel tree and feed it to an int err,
printing to kernel output when an error occurs.

Fixes: "AOSCOS: MIPS: Loongson 3: Add platform device drivers"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/platform/mips/lemote3a-laptop.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/platform/mips/lemote3a-laptop.c b/drivers/platform/mips/lemote3a-laptop.c
index 6e6b0c46aca4..a8d339a3e30b 100644
--- a/drivers/platform/mips/lemote3a-laptop.c
+++ b/drivers/platform/mips/lemote3a-laptop.c
@@ -542,10 +542,16 @@ static int lemote3a_laptop_suspend(struct platform_device * pdev, pm_message_t s
 /* Platform device resume handler */
 static int lemote3a_laptop_resume(struct platform_device * pdev)
 {
+	int err;
 	struct pci_dev *dev;
 
 	dev = pci_get_device(PCI_VENDOR_ID_ATI, PCI_DEVICE_ID_ATI_SBX00_SMBUS, NULL);
-	pci_enable_device(dev);
+	err = pci_enable_device(dev);
+	if (err) {
+		dev_err(&pdev->dev, "lemote3a-laptop: cannot enable ATI device: %d\n",
+			err);
+		return err;
+	}
 
 	/* Process LID event */
 	lemote3a_sci_event_handler(SCI_EVENT_NUM_LID);
-- 
2.48.1

